subprojects/core/src/main/java/org/gradle/util/NameValidator.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright 2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.util;
+
+import org.apache.commons.lang.StringUtils;
+
+import java.util.Arrays;
+
+public final class NameValidator {
```

Comment:
If we can test the same functionality with unit tests rather than integration tests that would be even better. Unit tests are usually faster to execute. Nevertheless, there has to be some integration tests but it can be lighter.